### ISSUE TITLE
Tokens on higher elevation appear to tokens on lower level of elevation

### ISSUE THREAD

**Issue by xavien12** [theripper93/Levels#112]
Levels Version: 3.5

Pf2e Version: 4.3.4

Describe the bug
As a DM when you select a NPC Token (with the vision enabled not checked under the tokens vision options) on a lower elevation, all tokens on higher levels of elevation can be seen. By going to the tokens on the higher elevation and disable and re-enable link to actor size under appearance the tokens from the higher levels disappear again from the lower levels, but can still be selected and glitch in and out when you select a token from a lower elevation but disappear when hovered over again.

To Reproduce
1. Have the actual version of the levels module installed.
2. Create a scene with different elevations. 
3. Put a tile as floor for the higher elevation.
4. Put a token on that floor.
5. Go to a lower elevation.
6. Put a token on that elevation.
8. Go back to the higher elevation floor and select the NPC see that the token on the lower elevation is not shown.
9. Go to a lower elevation and see that that the token in the higher elevation is shown.
10. Go to appearance on the higher elevation token and disable and re-enable link to actor size. 
11. Token on the higher elevation is no longer shown on the lower elevation. 
12. In addition when clicking on a tokens on the lower elevation, the tokens on the higher elevation will appear again on the lower level but disappear when hovered over.

Expected behavior
When you go to a level you should only see the token on that level and also not should only be able select the tokens on that level

Screenshots

high elevation
![strange tokens](https://user-images.githubusercontent.com/58941502/200163473-0c051ff7-e390-4709-ad5c-66eeece56603.png)

lower elevation
![strange tokens 2](https://user-images.githubusercontent.com/58941502/200163470-0024b34d-607d-46be-ad52-f2347aec0324.png)
token on the lower right had been changed by going to appearance and disable and re-enable link to actor size.

Browser:
Firefox

Foundry Version:
v10.288

**Comment by theripper93** [comment]
Can you verify if this happens on other systems? If the system is doing something strange with token sizes I can’t do much about it, if it’s a widespread issue I’ll see how to fix it

**Comment by xavien12** [comment]
I will try it out later with 5e and see what happens. I tried to bring it up to the pf2e developer though they closed the issue because its module specific

**Comment by theripper93** [comment]
The token size problem is definitely related to the system, in the sense that they don’t commit to the token data the size until it’s opened and closed, so I’m not providing a fix for that . For the other issue there might be something wrong there

**Comment by theripper93** [comment]
The token disappearing on hover side of the issue you were talking about should now be fixed and working as intended

**Comment by xavien12** [comment]
I tried it now with 5e and it works normally for that, but for pf2e the issue still persists. If I click on an NPC token on a lower elevation I see all tokens on higher elevation. The change that happened though was now even when I tried the temporary solution the upper elevation tokens stay visible when i select the lower elevation tokens. Though the temporary solution atleast hides the upper elevation when i just click on the layer without any tokens selected
